js/reference.js

```diff
@@ -53,6 +59,12 @@
             }
         }
 
+        // properly initialize facetPanelDisplay
+        var ccFacetDisplay = res.facetPanelDisplay;
```

Comment:
I don't think this makes sense. You shouldn't modify the properties inside the object. If the object is valid, we should use it without any modification.

If you define something like which is valid
```
{
  open: ["compact/select/association"]
}
```
Then this code is adding the `closed`, which doesn't make sense. 

If your goal is validation, you should validate each property and if all of them are invalid, then use the default.